File: SYP.Web/Modules/ServerTypes/Catalog/PriceListItemsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PriceListItemsRow } from "./PriceListItemsRow";

export interface PriceListItemsColumns {
    ProductCode: Column<PriceListItemsRow>;
    ProductName: Column<PriceListItemsRow>;
    UnitPrice: Column<PriceListItemsRow>;
    DiscountRate: Column<PriceListItemsRow>;
    Notes: Column<PriceListItemsRow>;
}

export class PriceListItemsColumns extends ColumnsBase<PriceListItemsRow> {
    static readonly columnsKey = 'Catalog.PriceListItems';
    static readonly Fields = fieldsProxy<PriceListItemsColumns>();
}

File: SYP.Web/Modules/Catalog/PriceListItems/PriceListItemsColumns.cs
using Serenity.ComponentModel;

namespace SYP.Catalog.Columns;

[ColumnsScript("Catalog.PriceListItems")]
[BasedOnRow(typeof(PriceListItemsRow), CheckNames = true)]
public class PriceListItemsColumns
{
    [EditLink, Width(100)]
    public string ProductCode { get; set; }

    [Width(250)]
    public string ProductName { get; set; }

    [Width(120), AlignRight]
    public decimal UnitPrice { get; set; }

    [Width(100), AlignRight]
    public decimal DiscountRate { get; set; }

    [Width(200)]
    public string Notes { get; set; }
}
